src/screens/PaymentRequestDetail.js

```diff
@@ -62,6 +69,18 @@ class PaymentRequestDetail extends React.Component {
     this.props.dispatch(clearInvoice());
   }
 
+  /**
+   * If the payment was not yet received, the user goes back to the "Select Amount" screen.
+   * Otherwise, goes directly to the Home screen to see the updated wallet.
+   */
+  onBackClick() {
+    if (this.state.paymentIsConfirmed) {
```

Comment:
Why can't you use `if (this.props.payment !== null)` instead of creating a new state?